pkg/monitor/datapath_trace_test.go

```diff
@@ -4,48 +4,125 @@
 package monitor
 
 import (
+	"bytes"
+	"encoding/binary"
+
 	. "gopkg.in/check.v1"
+
+	"github.com/cilium/cilium/pkg/byteorder"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/types"
 )
 
-func (s *MonitorSuite) TestDecodeTraceNotify(c *C) {
-	tn := &TraceNotify{}
-	err := DecodeTraceNotify([]byte{}, tn)
-	c.Assert(err, NotNil)
+func (s *MonitorSuite) TestDecodeTraceNotifyV0(c *C) {
+	// This check on the struct length constant is there to ensure that this
+	// test is updated when the struct changes.
+	c.Assert(traceNotifyV0Len, Equals, 32)
 
-	data := []byte{
-		0x00,       // Type
-		0x00,       // ObsPoint
-		0x00, 0x00, // Source
-		0x00, 0x00, 0x00, 0x00, // Hash
-		0x00, 0x00, 0x00, 0x00, // OrigLen
-		0x00, 0x00, // CapLen
-		0x00, 0x00, // Version
-		0x00, 0x00, 0x00, 0x00, // SrcLabel
-		0x00, 0x00, 0x00, 0x00, // DstLabel
-		0x00, 0x00, // DstID
-		0x00,                   // Reason
-		0x00,                   // Flags
-		0x02, 0x00, 0x00, 0x00, // Ifindex
+	fields := []any{
+		uint8(0x00),                              // Type
+		uint8(0x02),                              // ObsPoint
+		uint16(0x03_04),                          // Source
+		uint32(0x05_06_07_08),                    // Hash
+		uint32(0x09_0a_0b_0c),                    // OrigLen
+		uint16(0x0d_0e),                          // CapLen
+		uint16(TraceNotifyVersion0),              // Version
+		identity.NumericIdentity(0x_11_12_13_14), // SrcLabel
+		identity.NumericIdentity(0x_15_16_17_18), // DstLabel
+		uint16(0x19_1a),                          // DstID
+		uint8(0x1b),                              // Reason
+		uint8(0x1c),                              // Flags
+		uint32(0x1d_1e_1f_20),                    // Ifindex
 	}
 
-	err = DecodeTraceNotify(data, tn)
+	buf := bytes.NewBuffer(nil)
+	for _, v := range fields {
+		err := binary.Write(buf, byteorder.Native, v)
```

Comment:
We could probably should directly write a `TraceNotify` struct using `binary.Write` here, instead of assuming the order of fields in the test and writing them one by one. If the order of fields in `TraceNotify` were to change, this test would not catch it, I think.

You might even be able to use `testutils.MustCreateL3L4Payload` here.

Reply:
Ah yes good catch, will update!

Reply:
Updated as suggested, but couldn't use `testutils.MustCreateL3L4Payload` because of an import cycle, please take another look :pray: